lib/rmt/cli/mirror.rb

```diff
@@ -99,20 +102,20 @@ def mirror_repos!(mirror, repos)
       errors << _("Repository '%{repo_name}' (%{repo_id}): %{error_message}") % {
         repo_id: repo.id, repo_name: repo.name, error_message: e.message
       }
-    ensure
-      mirrored_repo_ids << repo.id
     end
-
-    [mirrored_repo_ids, errors]
   end
 
-  def handle_errors!(errors)
-    return if errors.empty?
-
-    raise RMT::CLI::Error.new(
-      _('The following errors ocurred while mirroring:%{errors_list}') % {
-        errors_list: "\n" + errors.join("\n")
-      }
-    )
+  def finish_execution
+    if errors.empty?
+      logger.info("\e[32m" + _('Mirroring complete.') + "\e[0m")
+    else
+      errors_list = errors.map { |e| e.end_with?('.') ? e : e + '.' }.join("\n")
+      raise RMT::CLI::Error.new(
+        "\e[33m" + _('Mirroring completed with errors.') + "\e[0m\n" +
+        _('The following errors ocurred while mirroring:%{errors_list}') % {
+          errors_list: "\n\e[31m" + errors_list + "\e[0m"
```

Comment:
While I ran this command, I saw that it could potentially use some improvement.

```suggestion
        "\e[31m" +
        _('The following errors occurred while mirroring:%{errors_list}') % {
          errors_list: "\n" + errors_list
        } +
      "\e[0m\n\e[33m" + _('Mirroring completed with errors.') + "\e[0m\n"
```

Notes:
* The "Mirroring completed with errors" works better as the final message, because it makes it clear that there is no more output expected.
* "The following errors occurred while mirroring" was easy to not associate with the error messages because of the difference in color.
* There was a typo in the word "occurred"

Reply:
Implemented the suggestions on a50305c.